freqtrade/maintenance_margin.py

```diff
@@ -0,0 +1,52 @@
+from typing import List
+
+from freqtrade.enums import TradingMode
+from freqtrade.leverage import liquidation_price
+from freqtrade.persistence import Trade
+
+
+class MaintenanceMargin:
+
+    trades: List[Trade]
+    exchange_name: str
+    trading_mode: TradingMode
+
+    @property
+    def margin_level(self):
+        # This is the current value of all assets,
+        # and if you pass below liq_level, you are liquidated
+        # TODO-lev: Add args to formula
+        return liquidation_price(
+            trading_mode=self.trading_mode,
+            exchange_name=self.exchange_name
+        )
+
+    @property
+    def liq_level(self):    # This may be a constant value and may not need a function
+        # TODO-lev: The is the value that you are liquidated at
+        return              # If constant, would need to be recalculated after each new trade
+
+    def __init__(self, exchange_name: str, trading_mode: TradingMode):
+        self.exchange_name = exchange_name
+        self.trading_mode = trading_mode
+        return
+
+    def add_new_trade(self, trade):
+        self.trades.append(trade)
+
+    def remove_trade(self, trade):
+        self.trades.remove(trade)
```

Comment:
I don't think this is a good way to handle this, as it'll be quite error-prone and work intensive (from a code maintenance perspective) to get this right.

This will work ONLY when a trade is handled within a bot run - but will fail the moment i stop and restart the bot with a trade open.
It'll also bear the risk that modifications to the trade will be missed (it's basically storing a copy of an object from the database - but we requery the database with every run - so we risk getting out of sync with this).


A safer approach (instead of keeping a reference around) is to query the database for relevant trades whenever required.
The required info must be in the database anyway, so it's as simple as writing an appropriate query (or better yet, query helper) to fetch relevant trades (open and have an applicable maintenance mode).

I'm not sure yet if that should be done within this class, or if a list of trades should be passed into `.run()` - basically doing the query where it's called (this might simplify handling of this during backtesting)